sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFramesSuite.scala

```diff
@@ -513,4 +513,14 @@ class DataFrameWindowFramesSuite extends QueryTest with SharedSparkSession {
       df,
       Row(1) :: Row(1) :: Nil)
   }
+
+  test("SPARK-45352: Remove window partition if partition expression are foldable") {
+    val ds = Seq(1, 2, 3).toDF("i")
+    val sortOrder = SortOrder($"i".expr, Ascending)
+    val window1 = new WindowSpec(Seq(), Seq(sortOrder), UnspecifiedFrame)
+    val window2 = new WindowSpec(Seq(lit(1).expr), Seq(sortOrder), UnspecifiedFrame)
```

Comment:
Please add more test cases:

- partitions only unfoldable.
- Mix unfoldable and foldable partition specs.

Reply:
@beliefer Thanks, updated all.